File: backend/app/schemas/lisa_context.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LisaConversationContextResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    organization_id: uuid.UUID
    user_id: uuid.UUID
    current_aircraft_id: uuid.UUID | None
    current_work_order_id: uuid.UUID | None
    current_task_id: uuid.UUID | None
    current_part_id: uuid.UUID | None
    current_part_requirement_id: uuid.UUID | None
    current_procurement_request_id: uuid.UUID | None
    current_vendor_id: uuid.UUID | None
    current_purchase_order_id: uuid.UUID | None
    current_technician_user_id: uuid.UUID | None
    current_aog_event_id: uuid.UUID | None
    previous_question: str | None
    context_version: int
    last_activity_at: datetime


class ResolvedEntityResponse(BaseModel):
    entity_type: str
    entity_id: str
    display: str

File: frontend/lib/api/lisaContext.ts
// Typed REAL-mode client for the backend Lisa conversation-context
// endpoints (backend/app/api/v1/lisa.py: GET/POST /lisa/context...).
// organization_id/user_id are never sent — the backend derives them from
// the authenticated JWT.

import { apiRequest } from "@/lib/apiClient";

export interface BackendLisaConversationContext {
  id: string;
  organization_id: string;
  user_id: string;
  current_aircraft_id: string | null;
  current_work_order_id: string | null;
  current_task_id: string | null;
  current_part_id: string | null;
  current_part_requirement_id: string | null;
  current_procurement_request_id: string | null;
  current_vendor_id: string | null;
  current_purchase_order_id: string | null;
  current_technician_user_id: string | null;
  current_aog_event_id: string | null;
  previous_question: string | null;
  context_version: number;
  last_activity_at: string;
}

export const lisaContextApi = {
  get: (accessToken: string) =>
    apiRequest<BackendLisaConversationContext>("/lisa/context", { accessToken }),
  reset: (accessToken: string) =>
    apiRequest<BackendLisaConversationContext>("/lisa/context/reset", {
      method: "POST",
      accessToken,
    }),
};
